### PR TITLE
Align http headers for parity

### PR DESCRIPTION
**Pull request by AsafMah**
x-ms-user-id
x-ms-app
x-ms-client-version

From either KCSB or CRP

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/client_base.py [hunk 1]
```diff
@@ -4,20 +4,23 @@
 import uuid
 from copy import copy
 from datetime import timedelta
-from typing import Union, Optional, Any, NoReturn, ClassVar
+from typing import Union, Optional, Any, NoReturn, ClassVar, TYPE_CHECKING
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client_base.py:7]: nit: did the linter go nuts on your PR?
If so, can we push a PR with forced linting for all files in the repository?

File: azure-kusto-data/azure/kusto/data/client_base.py [hunk 2]
```diff
@@ -48,10 +52,11 @@ def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str], is_async):
         self._request_headers = {
             "Accept": "application/json",
             "Accept-Encoding": "gzip,deflate",
-            "x-ms-client-version": "Kusto.Python.Client:" + VERSION,
             "x-ms-version": self.API_VERSION,
         }
 
+        self.client_details = self._kcsb.client_details
+
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client_base.py:59]:
```suggestion
```

File: azure-kusto-data/azure/kusto/data/client_base.py [hunk 3]
```diff
@@ -144,14 +149,39 @@ def __init__(
...
+        special_headers = [
+            {
+                "name": "x-ms-client-request-id",
+                "value": client_request_id_prefix + str(uuid.uuid4()),
+                "property": lambda p: p.client_request_id,
+            },
+            {
+                "name": "x-ms-client-version",
+                "value": client_details.version_for_tracing,
+                "property": lambda p: client_details.version_for_tracing,
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client_base.py:162]:
```suggestion
                "property": lambda p: None,
```

File: azure-kusto-data/azure/kusto/data/client_base.py [hunk 4]
```diff
@@ -144,14 +149,39 @@ def __init__(
...
+            },
+            {
+                "name": "x-ms-user",
+                "value": client_details.user_name_for_tracing,
+                "property": lambda p: p.user,
+            },
+        ]
+
+        for header in special_headers:
+            value = header["value"]
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client_base.py:177]: else?
(why set value twice?)

**Review by yogilad — APPROVED** [review]
Some suggestions in the comments

File: azure-kusto-data/azure/kusto/data/client_base.py (hunk 1 above)
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/client_base.py:7]: There was a new version of black, so that's probably what has changed

**Pull request merged**